js/ui/appIconBar.js

```diff
@@ -1011,6 +1064,20 @@ const AppIconBar = new Lang.Class({
         this._panel.closeActiveMenu();
     },
 
+    _activateNthApp: function(index) {
+        if (index == 0) {
+            this._browserButton.activateFirstWindow();
+            return;
+        }
+        this._scrolledIconList.activateNthApp(index - 1);
+    },
+
+    _activateLastApp: function() {
+        // Activate the index of the last button in the scrolled list + 1, to
+        // include the browser button
+        this._activateNthApp(this._scrolledIconList.getNumAppButtons());
```

Comment:
You could make it so that passing `-1` activates the last app. That would most likely remove the need for the `getNumAppButtons()` method on `scrolledIconList`